src/levanter/data/shard_cache.py

```diff
@@ -1737,6 +1737,10 @@ def unshard(self):
         """
         return ShardCache(self.cache_dir, self._batch_size, self._ledger, self._broker, 0, 1)
 
+    def set_batch_size(self, batch_size):
```

Comment:
i strongly prefer immutability by default (with an exception for iterators and performance). could we do with_batch_size and reutrn a copy?

Reply:
Yes - of course! As a former Scala dev, I should have defaulted to the immutable version :laughing:

Reply:
Now it makes sense why you figured all this out so quickly :)